observer/holder/nodesHolder_test.go

```diff
@@ -0,0 +1,200 @@
+package holder
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodesHolder_ConstructorAndGetters(t *testing.T) {
```

Comment:
Let's stop reinventing new ways of writing tests, especially for complex components. So, let's have a proper test for the constructor and for all setters & getters

Reply:
added a new test for the constructor. For getters, I preferred to keep the same test (split in 2: before and after update) because one getter's response is considered 'correct' if the other getter's response is also correct